src/foundry/client/apps/forms/journal-page-sheet.d.ts

```diff
@@ -0,0 +1,292 @@
+      limited: JournalEntryPage["limited"];
+      options?: Partial<T>;
+      owner: JournalEntryPage["isOwner"];
+      title: string;
+    }
+  }
+  /**
+   * The Application responsible for displaying and editing a single {@link JournalEntryPage} text document.
+   */
+  class JournalTextPageSheet<
+    TOptions extends JournalTextPageSheet.Options = JournalTextPageSheet.Options
+  > extends JournalPageSheet<TOptions> {
+    /**
+     * Bi-directional HTML \<-\> Markdown converter.
+     */
+    static get _converter(): Showdown.Converter;
+    /**
+     * Declare the format that we edit text content in for this sheet so we can perform conversions as necessary.
+     */
+    static get format(): JOURNAL_ENTRY_PAGE_FORMATS;
+    /**
+     * @defaultValue
+     * ```typescript
+     * const options = super.defaultOptions;
+     * options.classes.push("text");
+     * options.secrets.push({parentSelector: "section"});
+     * ```
+     */
+    static get defaultOptions(): JournalTextPageSheet.Options;
+
+    /**
+     * The table of contents for this JournalTextPageSheet.
+     */
+    toc: ReturnType<typeof JournalEntryPage["buildTOC"]>;
+
+    // FIXME: this should be async once the parent classes are updated
+    override getData(options?: Partial<TOptions> | undefined): JournalPageSheet.SheetData<TOptions>;
+
+    // FIXME: this should be async once the parent classes are updated
+    close(options?: FormApplication.CloseOptions | undefined): Promise<void>;
+
+    // FIXME: this should be async once the parent classes are updated
+    protected _render(
+      force?: boolean | undefined,
+      options?: Application.RenderOptions<TOptions> | undefined
+    ): Promise<void>;
+
+    /**
+     * Determine if any editors are dirty.
+     */
+    isEditorDirty(): boolean;
+
+    // FIXME: this should be async once the parent classes are updated
+    activateEditor(
+      name: string,
+      options?: TextEditor.Options | undefined,
+      initialContent?: string | undefined
+    ): Promise<Editor | EditorView>;
+    // FIXME: this should be async once the parent classes are updated
+    protected _renderInner(data: object): Promise<JQuery<HTMLElement>>;
+    _getSecretContent(secret: HTMLElement): string;
+    // FIXME: revisit this when it's added to its ancestor classes
+    _updateSecret(secret: HTMLElement, content: string): Promise<typeof this>;
+    // FIXME: this should be async once the parent classes are updated
+    protected _updateObject(event: Event, formData: object): Promise<unknown>;
+    /**
+     * Lazily convert text formats if we detect the document being saved in a different format.
+     * @param renderData - Render data.
+     */
+    protected _convertFormats(renderData: ReturnType<typeof this["getData"]>): void;
+
+    /* -------------------------------------------- */
+    /*  HTML Editing                                */
+    /* -------------------------------------------- */
```

Comment:
These free floating comments should be removed. It is the only thing we don't copy from foundry and its member ordering.